src/platform/plugins/shared/controls/common/options_list/types.ts

```diff
@@ -76,31 +86,53 @@ export type OptionsListResponse = OptionsListSuccessResponse | OptionsListFailur
 /**
  * The Options list request type taken in by the public Options List service.
  */
-export type OptionsListRequest = Omit<
-  OptionsListRequestBody,
-  'filters' | 'fieldName' | 'fieldSpec'
-> & {
+interface OptionsListRequestBase
+  extends Pick<OptionsListControlState, 'searchTechnique' | 'sort' | 'selectedOptions'> {
+  runtimeFieldMap?: Record<string, RuntimeFieldSpec>;
+  allowExpensiveQueries: boolean;
+  runPastTimeout?: boolean;
+  searchString?: string;
+  size: number;
+  ignoreValidations?: boolean;
   timeRange?: TimeRange;
-  dataView: DataView;
+  dataView?: DataView;
   filters?: Filter[];
-  field: FieldSpec;
+  field?: FieldSpec;
   query?: Query | AggregateQuery;
+}
+
+export type OptionsListDSLRequest = OptionsListRequestBase &
+  Required<Pick<OptionsListRequestBase, 'dataView' | 'field'>>;
+
+export type OptionsListESQLRequest = Omit<
```

Comment:
Is `OptionsListESQLRequest` type needed? ESQL does not cache requests in its current implemenation.